addons/point_of_sale/tests/test_point_of_sale_flow.py

```diff
@@ -1957,3 +1957,98 @@ def test_multi_exp_account_real_time(self):
 
         moves = self.env['account.move'].search([('ref', '=', f'pos_order_{order.id}')])
         self.assertEqual(len(moves), 2)
+
+    def test_multi_currency_journal(self):
+        """Verify that debit and credit are balanced for a company currency as EUR and a selling currency as USD."""
+
+        # Get currencies
+        currency_usd = self.env['res.currency'].search([('name', '=', 'USD')], limit=1)
+        currency_eur = self.env['res.currency'].search([('name', '=', 'EUR')], limit=1)
+        self.env.company.currency_id = currency_eur.id
+
+        # Set USD rate compared with EUR
+        currency_usd.rate_ids.create({
+            'company_rate': 2.0,
+            'currency_id': currency_usd.id,
+            'company_id': self.env.company.id
+        })
+
+        # Set USD to journals
+        self.env.company.bank_journal_ids.currency_id = currency_usd.id
+        journal_cash = self.env['account.journal'].search(
+            [('type', '=', 'cash'), ('company_id', '=', self.env.company.id)], limit=1
+        )
+        journal_cash.currency_id = currency_usd.id
+
+        journal_pos = self.env['account.journal'].create({
+            'name': 'Pos',
+            'type': 'general',
+            'code': 'POS',
+            'currency_id': currency_usd.id,
+            'company_id': self.env.company.id,
+        })
+        self.pos_config.journal_id = journal_pos.id  # Set the journal to the PoS
+
+        pos_account = self.env['account.account'].search(
+            [('company_id', '=', self.env.company.id), ('name', '=', 'Account Receivable (PoS)')], limit=1
+        )
+
+        # Set cash and bank payment methods
+        cash_payment_method = bank_payment_method = False
+        for pm in self.pos_config.payment_method_ids:
+            if pm.name in ('Cash', 'Bank'):
+                pm.receivable_account_id = pos_account.id
+                if pm.name == 'Bank':
+                    pm.journal_id = self.env.company.bank_journal_ids.id
+                    pm.outstanding_account_id = self.env['account.account'].search(
+                        [('company_id', '=', self.env.company.id), ('name', '=', 'Outstanding Receipts')], limit=1
+                    ).id
+                    bank_payment_method = pm
+                else:
+                    pm.journal_id = journal_cash.id
+                    cash_payment_method = pm
+
+        # Make 3 sales, 2 paid by cash and 1 by bank
+        self.pos_config.open_ui()
+        for payment_id, nb in zip((cash_payment_method.id, cash_payment_method.id, bank_payment_method.id), range(1, 4)):
```

Comment:
Do we really need to make 3 different orders ?

Reply:
No, however, we still need 1 paid by cash and 1 by bank. (applied in next push)